src/main/java/org/logstash/beats/Message.java

```diff
@@ -0,0 +1,61 @@
+package org.logstash.beats;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Message implements Comparable<Message> {
+    private final int sequence;
+    private String identityStream;
+    private final Map data;
```

Comment:
The problem here is the payload might not always be string in the context of the v2 protocol, filebeat. I think we rely in the code on jackson automatic understanding of json to create the right object, int, string or even other hash.